### ISSUE TITLE
USA Chinooks will discard move commands shortly after construction from USA Supply Center

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#217]
USA Chinooks will discard move commands shortly after construction from USA Supply Center. They will automatically move to Supply Stash instead.

**Comment by Jundiyy** [comment]
Already mentioned this before. It's the same for all 3 factions and their supply gatherers.

**Comment by Madrageeeeee** [comment]
Yes very annoying indeed and makes putting infantry inside for a drop very micro intensive

**Comment by ZekeDlyoung** [comment]
IIRC, it is a bugged caused by ChinookAI and can't really be fixed

**Comment by ImTimK** [comment]
A user can easily prevent this by pressing S (when it spawns) before making the move command. For people who made this a habit it never happens, but ofcourse it shouldn't be this way.

**Comment by fastAiRR** [comment]
Was checking to see if you had raised this. Its a very annoying bug. Stop command can stop it sure, but I don't think it always listens. Many times the thing flys away on me when im trying to get ready for a drop.

**Comment by kABUSE1** [comment]
As I pointed out in the duplicate issue , this behaviour makes setting a rally point without disabling auto-collect for a supply center possible, so you have to be cautious when fixing this.

Without a way to unassign a rally point from a building, you would permanently disable auto-collecting when you set a supply center to have a rally point.

**Comment by xezon** [comment]
Suggested functionality:

1. Supply Center waypoint moves the gatherer to location without gathering AI.
2. Waypoint is selectable and can be deleted with DEL button. Similar to how the Beacon works.

**Comment by MTKing4** [comment]
I think we have two options here

1. Disable Auto Gather and make waypoints useful for chinook drops/truck rushes/sending workers to locations, and add a way to clear waypoints, maybe wifh the stop command
2. Enable Auto Gather and fix the waypoints stoping the gathering after the first drop off.

**Comment by ImTimK** [comment]
If resource gatherers follow waypoints every time, manual collection will have to override this otherwise you can't collect from different piles which means functionality loss. 

It also means the order-dismissal-bug upon spawn (workers walking back to supplies instead of finishing scaffolds and chinooks flying back to supplies) needs to be 100 % fixable, otherwise this bug will occur much more often.

**Comment by xezon** [comment]
Not clear to me what this means. kabuse already explained things.

**Comment by MTKing4** [comment]
> If resource gatherers follow waypoints every time, manual collection will have to override this otherwise you can't collect from different piles which means functionality loss.

True but in that case for crates you just remove the waypoint with the hopefully implantable feature mentioned Above, and use waypoints only for supply docks, therefore no functionality is lost

> It also means the order-dismissal-bug upon spawn (workers walking back to supplies instead of finishing scaffolds and chinooks flying back to supplies) needs to be 100 % fixable, otherwise this bug will occur much more often.

If this bug is not fixed Auto Gather will be removed altogether
